### PR TITLE
Reformat functions, fix eslint and ts config

### PR DESCRIPTION
**Pull request by jeronimoek**
[Previous PR
](https://github.com/elchininet/ColorTranslator/pull/93)
Hi! I wanted to improve the project with a few changes:

- Reformatted a few functions (mainly in the src folder), and added some helpers.
- Modified the configuration for ESLint and Typescript. Added ignore file for ESLint
- Added file for ignoring Prettier in the whole project (useful to avoid new contributors formatting files accidentally)

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1 @@
+**
```
Comment (elchininet) [inline .prettierignore:1]: Please, remove this file, this project doesn‘t use Prettier. It‘s up to each contributor to avoid that their tools/set up mess things with the code but this package should not hold those configurations.

File: src/color/css.ts [hunk 2]
```diff
@@ -5,15 +5,36 @@ import {
     HSLObject,
     CMYKObject
 } from '@types';
-import {
-    toHEX,
-    hasProp,
-    round
-} from '#helpers';
+import { toHEX, hasProp, round, roundAll, toHEXAll } from '#helpers';
```
Comment (elchininet) [inline src/color/css.ts:8]: I prefer more than two imports in different lines, even if Prettier don‘t do it.

File: src/color/translators.ts [hunk 3]
```diff
@@ -156,16 +163,17 @@ export const hueRYB = (hue: number, toRYB: boolean): number => {
...
     let d = 0;
 
-    from.find((arr: [number, number], index: number): boolean => {
-        if (hue >= arr[0] && hue < arr[1]) {
-            a = arr[0];
-            b = arr[1];
-            c = to[index][0];
-            d = to[index][1];
-            return true;
+    for (let i = 0; i < from.length; i++) {
```
Comment (elchininet) [inline src/color/translators.ts:166]: What is the purpose of this change? If it is for legibility, I prefer the previous one.

File: src/helpers/index.ts [hunk 4]
```diff
@@ -54,14 +58,22 @@ export const getBase255Number = (color: string, alpha = false): number => {
 export const getCMYKNumber = (color: string): number => Math.min(PCENT.test(color) ? +(color.replace(PCENT, '$1')) / 100 : +color, 1);
 
 //---Return an ordered string from an array values
-export const getOrderedArrayString = (keys: string[]): string => keys.sort().join('').toUpperCase();
+export const getOrderedArrayString = (keys: string[]): string => {
+    const resultKeys = [...keys];
+    resultKeys.sort();
+    return resultKeys.join('').toUpperCase();
+};
```
Comment (elchininet) [inline src/helpers/index.ts:65]: If the intention is avoiding the array mutation, just spread the variable. It does‘t gain much legibility converting it to a block.

```suggestion
export const getOrderedArrayString = (keys: string[]): string =>
    [...keys].sort().join('').toUpperCase();
```

File: src/color/css.ts [hunk 5]
```diff
@@ -5,15 +5,36 @@ import {
...
+            l
+        )}%${alphaText})`;
+    },
+    [ColorModel.CMYK]: (color: CMYKObject): string => {
+        const { c, m, y, k } = color;
+        const values = roundAll([c, m, y, k]);
+        const hasAlpha = hasProp<CMYKObject>(color, 'a');
+        const alphaText = hasAlpha ? `,${round(color.a, MIN_DECIMALS)}` : '';
+        return `cmyk(${values.join('%,')}%${alphaText})`;
+    }
```
Comment (elchininet) [inline src/color/css.ts:39]: I like the refactoring of these methods 👍

File: src/color/translators.ts [hunk 6]
```diff
@@ -69,10 +69,7 @@ export const rgbToHSL = (r: number, g: number, b: number, a = 1): HSLObject => {
     let h = 0;
     let s = 0;
     const l = (max + min) / 2;
-    if (d === 0) {
-        h = 0;
-        s = 0;
-    } else {
+    if (d !== 0) {
```
Comment (elchininet) [inline src/color/translators.ts:72]: 👌

**Comment by jeronimoek** [comment]
Changes uploaded

**Review by elchininet — APPROVED** [review]

**Comment by elchininet** [comment]
@jeronimoek,
Merged.
Thanks for contributing 🙌
Regards

**Comment by jeronimoek** [comment]
Thank you!
